include/packet/superpacket.hpp

```diff
@@ -25,23 +27,27 @@
 
 class SuperPacket {
   public:
-    SuperPacket(void *pkt, uint32_t max_payload_len);
+    SuperPacket(void *pkt, uint32_t max_payload_len, Config *c);
```

Comment:
If we don't set the wireless / wired in the config and instead parse the packets, is this needed?

Reply:
But you actually used the config in the get_bitstring and get_index functions of SuperPacket class. So, I would suggest just set the config point in the Super Packet constructer.
 
https://github.com/nprint/nprint/blob/eb76192cb0d5dd606da479663d4782084c9f5148/include/packet/superpacket.hpp#L36-L37

Reply:
I think you're right here about passing the config in to the constructor instead of each individual function.